### DISCUSSION TITLE
[FEATURE]: Allow listing to be set to sold_out

### DISCUSSION THREAD

**Discussion by amitkumar-248** [etsy/open-api#718] [Ideas]
### Is there an existing issue for this?

- [X] I have searched the existing tickets

### Current Endpoint Design

currently, we don't have any option to put inventory as 0 on Etsy when it's out of stock in our warehouse. so we have to just put that listing in state inactive with at least an inventory of 1. But after some time when the seller gets back to their seller panel and sees some inventory(suppose 1) on that time client manually activates that (supposing it was deactivated by mistake, bcoz it's having 1 inventory). 

after 1 week or after a few days when the seller gets an order for that item and that order was not successfully processed due to out of stock issue. that time seller comes and says inventory syncing not working properly.   

### Proposed Endpoint Design Change

proposing to add one more index in [getListing](https://developers.etsy.com/documentation/reference/#operation/getListing) and 
[getListingInventory](https://developers.etsy.com/documentation/reference/#operation/getListingInventory) as 
'api_gateway' => enum (Etsy, 3rd_party_app, API_KEYSTRING, etc.) 

### Why are you proposing this change?

after adding the above endpoint we will be able to be 100% sure whether this listing is directly activated/modified from Etsy or by any other medium.  

hoping to get this index added, 
thanks in advance @etsyachristensen

**Comment by etsyachristensen** [comment]
For the inventory issue we have a ticket already for allowing sold_out status. We just haven't been able to work out a way to make this happen.

I'm not following the purpose of the requested "index". What would this field be for?

**Comment by etsyachristensen** [comment]
We're converting this to an Idea discussion for continuing tracking.

**Comment by jsalachna-bl** [comment]
@etsyachristensen what is status of:
"For the inventory issue we have a ticket already for allowing sold_out status. We just haven't been able to work out a way to make this happen."

**Comment by josephspurrier** [comment]
@rzunigaca @simpsonw (tagging for some visibility) It would also be good to confirm if these discussion boards are still monitored by Etsy.

Yes, agreed, would love to have the status of this. It's been open since 2022 and would love to ensure the customers have the best experience when working with our stores on items that will be back in store.

**Reply by alexh-swdev** [reply]
+1
